examples/bsoncxx/builder_list.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/builder/list.hpp>
+#include <bsoncxx/types.hpp>
+#include <bsoncxx/types/bson_value/value.hpp>
+#include <chrono>
+
+using namespace bsoncxx;
+
+int main(int, char**) {
+    using namespace bsoncxx::builder;
+
+    //
+    // bsoncxx::builder::list, bsoncxx::builder::document, and bsoncxx::builder::array provides a
+    // JSON-like interface for creating BSON objects.
+    //
+
+    // builder::document builds an empty BSON document
+    builder::document doc = {};
+    // builder::array builds an empty BSON array
+    builder::array arr = {};
+
+    //
+    // We can append values to a document using an initializer list of key-value pairs.
+    // { "hello" : "world" }
+    //
+    doc = {"hello", "world"};
+
+    //
+    //  Each document key must be a string type.
+    //
+    // {
+    //   "c-style" : "with value",
+    //   "basic string" : "with value"
+    // }
+    //
+    doc = {"c-style", "with value", std::string("basic string"), "with value"};
+
+    //
+    // Each document value must be a bson_value::value or implicitly convertible to one.
+    //
+    // {
+    //   "BSON boolean value" : false,
+    //   "BSON 32-bit signed integer value" : -123,
+    //   "BSON date value" : { "$date" : 123456789 },
+    //   "BSON Decimal128 value" : { "$numberDecimal" : "1.844674407370955161800E-6155" },
+    //   "BSON regex value with options" : { "$regex" : "any", "$options" : "imsx" }
+    //  }
+    //
+    // clang-format off
```

Comment:
clang would put each key and each value below on its own line, obfuscating the value description in the key.